utils/responses.ts

```diff
@@ -0,0 +1,127 @@
+import * as express from "express";
+import * as t from "io-ts";
+import { errorsToReadableMessages } from "italia-ts-commons/lib/reporters";
+import { IResponseType } from "italia-ts-commons/lib/requests";
+import {
+  HttpStatusCodeEnum,
+  IResponse,
+  IResponseErrorForbiddenNotAuthorized,
+  IResponseErrorInternal,
+  IResponseErrorNotFound,
+  IResponseErrorTooManyRequests,
+  ResponseErrorForbiddenNotAuthorized,
+  ResponseErrorGeneric,
+  ResponseErrorInternal,
+  ResponseErrorNotFound,
+  ResponseErrorTooManyRequests,
+  ResponseErrorValidation
+} from "italia-ts-commons/lib/responses";
+
+/**
+ * Interface for a no content response returning a empty object.
+ */
+export interface IResponseNoContent extends IResponse<"IResponseNoContent"> {
+  readonly value: {};
+}
+/**
+ * Returns a no content json response.
+ */
+export function ResponseNoContent(): IResponseNoContent {
+  return {
+    apply: (res: express.Response) => res.status(204).json({}),
+    kind: "IResponseNoContent",
+    value: {}
+  };
+}
+
+/**
+ * Transforms async failures into internal errors
+ */
+export const withCatchAsInternalError = <T>(
+  f: () => Promise<T>,
+  message: string = "Exception while calling upstream API (likely a timeout)."
+) =>
+  f().catch(_ => {
+    // tslint:disable-next-line:no-console
+    console.error(_);
+    return ResponseErrorInternal(`${message} [${_}]`);
+  });
+
+export const unhandledResponseStatus = (status: number) =>
+  ResponseErrorInternal(`unhandled API response status [${status}]`);
+
+/**
+ * Calls the provided function with the valid response, or else returns an
+ * IResponseErrorInternal with the validation errors.
+ */
+export const withValidatedOrInternalError = <T, U>(
+  validated: t.Validation<T>,
+  f: (t: T) => U
+) =>
+  validated.isLeft()
+    ? ResponseErrorInternal(
+        errorsToReadableMessages(validated.value).join(" / ")
+      )
+    : f(validated.value);
+
+/**
+ * Calls the provided function with the valid response, or else returns an
+ * IResponseErrorValidation with the validation errors.
+ */
+export const withValidatedOrValidationError = <T, U>(
```

Comment:
do we use these methods?

Reply:
Fixed in [98db5c7](https://github.com/pagopa/io-functions-services/pull/68/commits/98db5c77eefe2f2bbce99559596e00feed83aa0c)